### PR TITLE
Introduced 'deprecated' for actions, events, arguments and object properties

### PR DESCRIPTION
**Pull request by wilzbach**
Fixes https://github.com/microservices/microservice.guide/issues/57

### CODE REVIEW

File: schema/actions/README.md [hunk 1]
```diff
@@ -74,6 +74,9 @@ Within a named `action`, the following fields are available:
     "http | rpc | format": {
         "desc": "How to call the action. [Read more](/schema/interface/)",
         "required": true
+    },
+    "deprecated": {
+        "desc": "Whether this action is deprecated or not. The default value for this is `false`"
```
Comment (StoryscriptHuman) [inline schema/actions/README.md]: While a boolean *works*, the more I think about it, the more I feel we should follow the gql spec way.
`deprecated` isn't a thing and `deprecationReason` if set, tells people WHY it's deprecated and how to upgrade. If set, a field is considered deprecated

Comment (matthewhudson) [reply schema/actions/README.md]: @StoryscriptHuman Agreed.

Comment (wilzbach) [reply schema/actions/README.md]: How about we just set `deprecated` to be a `string` (with the same properties)?

Comment (matthewhudson) [reply schema/actions/README.md]: Perfect, sounds good to me 👍

Comment (wilzbach) [reply schema/actions/README.md]: Changed `deprecated` to accept a message string.

**Pull request merged**